From 4bea1f9402568434bbb95421d9da7cfe8fb23839 Mon Sep 17 00:00:00 2001
From: Tao Yang <Tao.Yang1@nxp.com>
Date: Tue, 26 May 2026 19:43:26 +0900
Subject: [PATCH] rteval: Tailored for NXP boards and Yocto rootfs

- Remove DMI info collection (not available on ARM boards)
- Remove sysreport/sosreport (not available in Yocto)
- Remove NetworkInfo from report (reduces dependencies)
- Change kcompile from allmodconfig to defconfig (ARM boards
  have limited memory)
- Use /etc/os-release for OS detection instead of redhat/fedora-release
- Remove LANG env variable from service status check (not set in Yocto)
- Remove -tt flag from shebang (deprecated in Python 3)

Upstream-Status: Inappropriate [embedded platform specific]
---
 rteval-cmd                       |  7 +------
 rteval/__init__.py               |  3 ---
 rteval/modules/loads/kcompile.py |  2 +-
 rteval/rteval_text.xsl           | 16 ----------------
 rteval/sysinfo/__init__.py       | 13 ++-----------
 rteval/sysinfo/osinfo.py         | 14 +++++++-------
 rteval/sysinfo/services.py       |  2 +-
 7 files changed, 12 insertions(+), 45 deletions(-)

diff --git a/rteval-cmd b/rteval-cmd
index 45d9f3a..73e5669 100755
--- a/rteval-cmd
+++ b/rteval-cmd
@@ -1,4 +1,4 @@
-#!/usr/bin/python3 -tt
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 # SPDX-License-Identifier: GPL-2.0-or-later
 #
@@ -117,9 +117,6 @@ def parse_options(cfg, parser, cmdargs):
     parser.add_argument("-i", "--installdir", dest="rteval___installdir",
                       type=str, default=rtevcfg.installdir, metavar="DIRECTORY",
                       help=f"place to locate installed templates (default: {rtevcfg.installdir})")
-    parser.add_argument("-s", "--sysreport", dest="rteval___sysreport",
-                      action="store_true", default=rtevcfg.sysreport,
-                      help=f'run sysreport to collect system data (default: {rtevcfg.sysreport})')
     parser.add_argument("-D", '--debug', dest='rteval___debugging',
                       action='store_true', default=rtevcfg.debugging,
                       help=f'turn on debug prints (default: {rtevcfg.debugging})')
@@ -184,7 +181,6 @@ def parse_options(cfg, parser, cmdargs):
     return cmd_opts
 
 if __name__ == '__main__':
-    from rteval.sysinfo import dmi
 
     # set LD_BIND_NOW to resolve shared library symbols
     # note: any string will do, nothing significant about 'rteval'
@@ -430,7 +426,6 @@ if __name__ == '__main__':
      debugging: {rtevcfg.debugging}
      logging:  {rtevcfg.logging}
      duration: {rtevcfg.duration}
-     sysreport: {rtevcfg.sysreport}''')
 
         if not os.path.isdir(rtevcfg.workdir):
             raise RuntimeError(f"work directory {rtevcfg.workdir} does not exist")
diff --git a/rteval/__init__.py b/rteval/__init__.py
index 6097ddf..077225d 100644
--- a/rteval/__init__.py
+++ b/rteval/__init__.py
@@ -251,11 +251,8 @@ class RtEval(rtevalReport):
             return 1
 
         self._report(measure_start, self.__rtevcfg.xslt_report)
-        if self.__rtevcfg.sysreport:
-            self._sysinfo.run_sysreport(self.__reportdir)
 
         if EARLYSTOP:
             rtevalres = 1
-        self._sysinfo.copy_dmesg(self.__reportdir)
         self._tar_results()
         return rtevalres
diff --git a/rteval/modules/loads/kcompile.py b/rteval/modules/loads/kcompile.py
index 404b46d..c96e40b 100644
--- a/rteval/modules/loads/kcompile.py
+++ b/rteval/modules/loads/kcompile.py
@@ -54,7 +54,7 @@ class KBuildJob:
             self.jobs = self.calc_jobs_per_cpu() * cpus_available
 
         self.runcmd = f"make O={self.objdir} -C {self.kdir} -j{self.jobs}"
-        self.cleancmd = f"make O={self.objdir} -C {self.kdir} clean allmodconfig"
+        self.cleancmd = f"make O={self.objdir} -C {self.kdir} clean defconfig"
         self.cleancmd += f"&& pushd {self.objdir} && {self.kdir}/scripts/config -d CONFIG_MODULE_SIG_SHA1 -e CONFIG_MODULE_SIG_SHA512 && popd && make O={self.objdir} -C {self.kdir} olddefconfig"
         if self.binder:
             self.runcmd = self.binder + " " + self.runcmd
diff --git a/rteval/rteval_text.xsl b/rteval/rteval_text.xsl
index 1463027..47d660f 100644
--- a/rteval/rteval_text.xsl
+++ b/rteval/rteval_text.xsl
@@ -45,22 +45,6 @@
     <xsl:value-of select="SystemInfo/uname/node|uname/node"/>
     <xsl:text>&#10;</xsl:text>
 
-    <xsl:text>   Model:        </xsl:text>
-    <xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='System Information']/Field[@Name='Manufacturer']"/>
-    <xsl:text> - </xsl:text><xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='System Information']/Field[@Name='Product Name']"/>
-    <xsl:text>&#10;</xsl:text>
-
-    <xsl:text>   BIOS version: </xsl:text>
-    <xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='BIOS Information']/Field[@Name='Vendor']"/>
-    <xsl:text> (ver: </xsl:text>
-    <xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='BIOS Information']/Field[@Name='Version']"/>
-    <xsl:text>, rev: </xsl:text>
-    <xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='BIOS Information']/Field[@Name='BIOS Revision']"/>
-    <xsl:text>, release date: </xsl:text>
-    <xsl:value-of select="SystemInfo/DMIinfo/Handle[@name='BIOS Information']/Field[@Name='Release Date']"/>
-    <xsl:text>)</xsl:text>
-    <xsl:text>&#10;&#10;</xsl:text>
-
     <xsl:text>   CPU cores:    </xsl:text>
     <xsl:choose>
       <xsl:when test="SystemInfo/CPUtopology">
diff --git a/rteval/sysinfo/__init__.py b/rteval/sysinfo/__init__.py
index 4b7b03c..ea2cd29 100644
--- a/rteval/sysinfo/__init__.py
+++ b/rteval/sysinfo/__init__.py
@@ -13,27 +13,20 @@ from rteval.sysinfo.services import SystemServices
 from rteval.sysinfo.cputopology import CPUtopology
 from rteval.sysinfo.memory import MemoryInfo
 from rteval.sysinfo.osinfo import OSInfo
-from rteval.sysinfo.newnet import NetworkInfo
 from rteval.sysinfo.cmdline import cmdlineInfo
 from rteval.sysinfo.tuned import TunedInfo
-from rteval.sysinfo import dmi
 
-class SystemInfo(KernelInfo, SystemServices, dmi.DMIinfo, CPUtopology,
-                 MemoryInfo, OSInfo, NetworkInfo, cmdlineInfo, TunedInfo):
+class SystemInfo(KernelInfo, SystemServices, CPUtopology,
+                 MemoryInfo, OSInfo, cmdlineInfo, TunedInfo):
     def __init__(self, config, logger=None):
         self.__logger = logger
         KernelInfo.__init__(self, logger=logger)
         SystemServices.__init__(self, logger=logger)
-        dmi.DMIinfo.__init__(self, logger=logger)
         CPUtopology.__init__(self)
         OSInfo.__init__(self, logger=logger)
         cmdlineInfo.__init__(self, logger=logger)
-        NetworkInfo.__init__(self, logger=logger)
         TunedInfo.__init__(self, logger=logger)
 
-        # Parse initial DMI decoding errors
-        self.ProcessWarnings()
-
         # Parse CPU info
         CPUtopology._parse(self)
 
@@ -45,11 +38,9 @@ class SystemInfo(KernelInfo, SystemServices, dmi.DMIinfo, CPUtopology,
         # Populate the report
         report_n.addChild(OSInfo.MakeReport(self))
         report_n.addChild(KernelInfo.MakeReport(self))
-        report_n.addChild(NetworkInfo.MakeReport(self))
         report_n.addChild(SystemServices.MakeReport(self))
         report_n.addChild(CPUtopology.MakeReport(self))
         report_n.addChild(MemoryInfo.MakeReport(self))
-        report_n.addChild(dmi.DMIinfo.MakeReport(self))
         report_n.addChild(cmdlineInfo.MakeReport(self))
         report_n.addChild(TunedInfo.MakeReport(self))
 
diff --git a/rteval/sysinfo/osinfo.py b/rteval/sysinfo/osinfo.py
index 3bdbcc8..90e908b 100644
--- a/rteval/sysinfo/osinfo.py
+++ b/rteval/sysinfo/osinfo.py
@@ -20,13 +20,13 @@ class OSInfo:
     def get_base_os(self):
         '''record what userspace we're running on'''
         distro = "unknown"
-        for f in ('redhat-release', 'fedora-release'):
-            p = os.path.join('/etc', f)
-            if os.path.exists(p):
-                f = open(p, 'r')
-                distro = f.readline().strip()
-                f.close()
-                break
+        p = os.path.join('/etc', 'os-release')
+        if os.path.exists(p):
+            with open(p, 'r') as f:
+                for line in f:
+                    if 'PRETTY_NAME' in line:
+                        distro = line.split('=', 1)[1].strip(' "\n')
+                        break
         return distro
 
 
diff --git a/rteval/sysinfo/services.py b/rteval/sysinfo/services.py
index c3c1553..736f268 100644
--- a/rteval/sysinfo/services.py
+++ b/rteval/sysinfo/services.py
@@ -46,7 +46,7 @@ class SystemServices:
                 c = subprocess.Popen(cmd, shell=True, encoding='utf-8')
                 c.wait()
                 if c.returncode == 0:
-                    cmd = ['env', '-i', f'LANG="{os.environ["LANG"]}"', f'PATH="{os.environ["PATH"]}"', f'TERM="{os.environ["TERM"]}"', service, 'status']
+                    cmd = ['env', '-i', f'PATH="{os.environ["PATH"]}"', f'TERM="{os.environ["TERM"]}"', service, 'status']
                     c = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
                     c.wait()
                     if c.returncode == 0 and (c.stdout.read() or c.stderr.read()):
-- 
2.43.0

